### PR TITLE
Support AgentPortrayalStyle in Altair visualization components

### PR DESCRIPTION
**Pull request by EwoutH**
### Problem
The Altair visualization components (`altair_components.py`) were not updated to support the new `AgentPortrayalStyle` class introduced in the visualization refactoring. When users returned an `AgentPortrayalStyle` instance from their `agent_portrayal` function, the code attempted dictionary-style assignment (`agent_data["x"] = x`), causing a `TypeError: 'AgentPortrayalStyle' object does not support item assignment`.

This also triggered a deprecation warning in tests that were still using the old dict-based portrayal format.

### Changes
- Added `_portrayal_to_dict()` helper function to convert `AgentPortrayalStyle` instances to dict format for Altair chart generation
- Updated `_get_agent_data_old__discrete_space()`, `_get_agent_data_new_discrete_space()`, and `_get_agent_data_continuous_space()` to use the helper function
- Updated `test_call_space_drawer` to use `AgentPortrayalStyle` instead of deprecated dict format
- Maintains full backward compatibility with existing dict-based portrayals

### Technical Details
Altair requires dict-formatted data for chart generation and only uses `color` and `size` fields from the agent portrayal (other fields become tooltips). The helper function extracts these fields from `AgentPortrayalStyle` instances while preserving the original dict-based behavior for legacy code.

Follow-up on https://github.com/mesa/mesa/pull/2786.

### CODE REVIEW

File: tests/test_solara_viz.py [hunk 1]
```diff
@@ -129,7 +130,7 @@ def __init__(self, seed=None):
     model = MockModel()
 
     def agent_portrayal(agent):
-        return {"marker": "o", "color": "gray"}
+        return AgentPortrayalStyle(marker="o", color="gray")
```
Comment (Sahil-Chhoker) [inline tests/test_solara_viz.py:133]: This is an old file, only kept for backwards compatibility, the updated file has all the necessary tests so you can leave this out of this PR.

Comment (EwoutH) [reply tests/test_solara_viz.py:133]: Right, so the correct behavior is to emit an warning here.

But then isn't it weird that this edge case in `altair` behavior was only catched by the old test file and not the new one?

Comment (Sahil-Chhoker) [reply tests/test_solara_viz.py:133]: This file contains the test for legacy backends, that's why it's kept, so it makes sense that it caught the error not the newer one.

Comment (EwoutH) [reply tests/test_solara_viz.py:133]: But is Altair tested at all on this aspect in the new test set? Was it validated it tested all new behavior? Because it started to fail when I tried to resolve a deprecation warning by updating old to new behavior in the old test set. The new one so apparently didn't include it.

Comment (Sahil-Chhoker) [reply tests/test_solara_viz.py:133]: You're right, older `altair` doesn't have any other tests, let's update it and can you rename the file as well to `test_solara_viz_legacy_backends` or something along those lines

Comment (EwoutH) [reply tests/test_solara_viz.py:133]: Sounds good!
